zcash_test_vectors/orchard_zsa/note_encryption.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+import sys;
+
+from .asset_base import native_asset
+
+assert sys.version_info[0] >= 3, "Python 3 required."
+
+from chacha20poly1305 import ChaCha20Poly1305
+from hashlib import blake2b
+
+from ..output import render_args, render_tv
+from ..rand import Rand
+
+from ..orchard.pallas import Point, Scalar
+from .commitments import value_commit
+from ..orchard.commitments import rcv_trapdoor
+from ..orchard.key_components import diversify_hash, prf_expand, FullViewingKey, SpendingKey
+from..orchard.note_encryption import kdf_orchard, prf_ock_orchard, OrchardKeyAgreement, OrchardSym, TransmittedNoteCipherText, OrchardNoteEncryption
+from .note import OrchardZSANote, OrchardZSANotePlaintext
+from ..orchard.utils import to_scalar
+
+
+# https://zips.z.cash/protocol/nu5.pdf#saplingandorchardencrypt
+class OrchardZSANoteEncryption(OrchardNoteEncryption):
+    def __init__(self, rand):
+        super().__init__(rand)
+
+    def encrypt(self, note: OrchardZSANote, memo, pk_d_new, g_d_new, cv_new, cm_new, ovk=None):
+        tc = super().encrypt(note, memo, pk_d_new, g_d_new, cv_new, cm_new, ovk)
```

Comment:
Although `super.encrypt(...)` takes `note: OrchardNote`, since `OrchardZSANote` inherits `OrchardNote`, if I pass it an `OrchardZSANote` instead, it will use the `OrchardZSANote` version of the functions wherever they differ. I had seen simpler examples online displaying that.

I first confirmed we get the longer length ciphertexts, which is an initial sanity check. 
Also then, I used these test vectors in the `orchard` crate with Dmitry's backward compatibility work, and it passed the test for `note_encryption_zsa` so it should be fine.